src/Explorer/Notebook/NotebookUtil.ts

```diff
@@ -100,4 +100,30 @@ export class NotebookUtil {
     const basePath = path.split(contentName).shift();
     return `${basePath}${newName}`;
   }
+
+  public static findFirstCodeCellWithDisplay(notebookObject: ImmutableNotebook): number {
+    let codeCellCount = -1;
+    for (let i = 0; i < notebookObject.cellOrder.size; i++) {
+      const cellId = notebookObject.cellOrder.get(i);
+      if (cellId) {
+        const cell = notebookObject.cellMap.get(cellId);
+        if (cell && cell.cell_type === "code") {
```

Comment:
This works. You might want to check the nteract's `selector` package to see if they have an iterator that saves you from doing all this access/check-if-exist logic.
Also, you can use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#:~:text=%20%20%20%20Desktop%20%20%20,74%20%20%20IE%20No%20support%20No%20) to remove some of the `if(blah)` conditions.

Reply:
I tried that, compile errors were thrown since notebookContent is set to unknown type in Explorer.ts since we did not want to have any nteract dependencies

Reply:
will add some of these changes to next PR